### PR TITLE
Schedule command extension

### LINKED ISSUE #93: Make parseClassTime prevent clashing class times
When parsing a class time to a student, check if the class time overlaps with any student in the model.

### PR DESCRIPTION
**Pull request by AlexCQY**
### View lesson schedule feature

UPDATED: Schedule now only supports viewing with 1 command. "Schedule view mode/[weekly/daily] date/[yyyy-mm-dd]"

1.Model

-Add  `Scheduler`, this is the main engine of scheduling and handles all operations of Events.
-Add `Mapper` for mapping of `Event` to `VEvent` for display in jfxtras iCalendar. All mapping are done in the `Scheduler`.

3.Logic

- Add `ScheduleViewCommand`. Supports viewing schedule weekly/daily and on a specific date. Default view is current week.

4.UI

- Add `SchedulePanel` for displaying schedule in a timetable. Library jfxtras is used here. 

* Added checks to prevent adding or editing a classTime to clash with current classTimes

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ScheduleCommandParser.java [hunk 1]
```diff
@@ -1,44 +1,49 @@
...
      * @throws ParseException if the user input does not conform the expected date format
-     * and the input is an empty string
      */
     @Override
-    public ScheduleCommand parse(String userInputDate) throws ParseException {
-        requireNonNull(userInputDate);
-        String trimmedUserInput = userInputDate.trim();
+    public ScheduleCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        requireNonNull(args);
```
Comment (VaishakAnand) [inline src/main/java/seedu/address/logic/parser/ScheduleCommandParser.java]: Duplicate line

**Review by VaishakAnand — COMMENTED** [review]
Looks good Alex, I believe you just duplicated one line wrongly, where I commented.
LGTM after that fix.

File: src/main/java/seedu/address/logic/parser/ScheduleCommandParser.java (hunk 1 above)
Comment (AlexCQY) [reply src/main/java/seedu/address/logic/parser/ScheduleCommandParser.java]: THANKS! I removed it already.

**Review by hogantan — APPROVED** [review]
lgtm

**Pull request merged**